### PR TITLE
Support for SWMM 5.1.12 and exposed error and engine version properties

### PR DESCRIPTION
**Pull request by bemcdonnell**
Addresses:
#85 #84 #83

### CODE REVIEW

File: pyswmm/simulation.py [hunk 1]
```diff
@@ -177,6 +177,77 @@ def execute(self):
         self._model.swmmExec()
 
     @property
+    def engine_version(self):
+        """
+        Retrieves the SWMM Engine Version
```
Comment (goanpeca) [inline pyswmm/simulation.py]: ` Retrieves the SWMM Engine Version.`

File: pyswmm/simulation.py [hunk 2]
```diff
@@ -177,6 +177,77 @@ def execute(self):
...
+        >>> sim.engine_version
+        StrictVersion("5.1.13")
+        
+        """
+        return self._model.swmm_getVersion()
+
+    @property
+    def runoff_error(self):
+        """
+        Retrieves the Runoff Mass Balance Error
```
Comment (goanpeca) [inline pyswmm/simulation.py]: `Retrieves the Runoff Mass Balance Error.`

File: pyswmm/simulation.py [hunk 3]
```diff
@@ -177,6 +177,77 @@ def execute(self):
...
+        >>> sim.runoff_error
+        0.01
+        
+        """
+        return self._model.swmm_getMassBalErr()[0]
+
+    @property
+    def flow_routing_error(self):
+        """
+        Retrieves the Flow Routing Mass Balance Error
```
Comment (goanpeca) [inline pyswmm/simulation.py]: `Retrieves the Flow Routing Mass Balance Error.`

File: pyswmm/simulation.py [hunk 4]
```diff
@@ -177,6 +177,77 @@ def execute(self):
...
+        >>> sim.flow_routing_error
+        0.01
+        
+        """
+        return self._model.swmm_getMassBalErr()[1]
+
+    @property
+    def quality_error(self):
+        """
+        Retrieves the Quality Routing Mass Balance Error
```
Comment (goanpeca) [inline pyswmm/simulation.py]: `Retrieves the Quality Routing Mass Balance Error.`

File: pyswmm/swmm5.py [hunk 5]
```diff
@@ -444,7 +450,10 @@ def getSimulationDateTime(self, timeType):
         errcode = self.SWMMlibobj.swmm_getSimulationDateTime(
             ctypes.c_int(timeType), dtme)
         self._error_check(errcode)
-        return datetime.strptime(dtme.value.decode("utf-8"), "%b-%d-%Y %H:%M:%S")
+        if self.swmm_getVersion() < distutils.version.StrictVersion('5.1.11'):
```
Comment (goanpeca) [inline pyswmm/swmm5.py]: `'5.1.11'` this number should go somewhere else, it feels like hardcoding stuff in a "hard to find" place

File: pyswmm/swmm5.py [hunk 6]
```diff
@@ -960,7 +969,10 @@ def getCurrentSimualationTime(self):
         dtme = ctypes.create_string_buffer(61)
         errcode = self.SWMMlibobj.swmm_getCurrentDateTimeStr(dtme)
         self._error_check(errcode)
-        return datetime.strptime(dtme.value.decode("utf-8"), "%b-%d-%Y %H:%M:%S")
+        if self.swmm_getVersion() < distutils.version.StrictVersion('5.1.11'):
```
Comment (goanpeca) [inline pyswmm/swmm5.py]: Same here

File: pyswmm/tests/test_general_engine.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import pyswmm
+from pyswmm import Simulation
+from pyswmm.tests.data import MODEL_WEIR_SETTING_PATH
+
+
+def test_engine_version():
+    pyswmm.lib.use("swmm5.dll")
+    sim = Simulation(MODEL_WEIR_SETTING_PATH)
+    print(sim.engine_version)
+    assert sim.engine_version == sim.engine_version
```
Comment (goanpeca) [inline pyswmm/tests/test_general_engine.py]: What kind of test is this :-p ?

**Comment by goanpeca** [comment]
@bemcdonnell are you using ciocheck?

**Comment by bemcdonnell** [comment]
@goanpeca, ciocheck has never worked for me ;) you said you were going to fix it

**Comment by goanpeca** [comment]
@bemcdonnell I assumed you started using python 3... and with that version I think works just fine

File: pyswmm/tests/test_general_engine.py (hunk 7 above)
Comment (bemcdonnell) [inline pyswmm/tests/test_general_engine.py]: Haaa meant to check a type against the type StrictVersion

**Comment by bemcdonnell** [comment]
I'm making my migration to py3 :-D

**Pull request merged**